src/Rotations.jl

```diff
@@ -0,0 +1,148 @@
+@doc doc"""
+    Rotations{N} <: Manifold
+
+Special orthogonal manifold $\mathrm{SO}(n)$ represented by $n\times n$
+real-valued orthogonal matrices with determinant $+1$.
+
+# Constructor
+
+    Rotations(n, retract! = rectract_qr!)
+
+generates the $\mathrm{SO}(n) \subset \mathbb R^{n\times n}$
+"""
+struct Rotations{N, TRetr<:Function} <: Manifold
+    retract!::TRetr
+end
+Rotations(n::Int, retract!::TRetr = retract_qr!) where TRetr = Rotations{n,TRetr}(retract!)
+
+@doc doc"""
+    dimension(S::Rotations)
+
+returns the dimension of the manifold $\mathrm{SO}(n)$, i.e. $\frac{n(n-1), 2}$.
+"""
+dimension(S::Rotations{N}) where {N} = div(N*(N-1), 2)
```

Comment:
I think we renamed it to manifold_dimension – but maybe we should just start a proper documentation with default names somewhen and then unify it globally again.

Reply:
I'll make a test for `manifold_dimension`.